src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -352,6 +352,25 @@ internal static bool GetUseLegacyDependencyResolver(IReadOnlyList<IVsTargetFrame
                 }
                 : null;
 
+            // For multi-targeting projects, we want to set audit mode to "all" if the project targets .NET 10 or higher.
+            // NuGet.targets achieves this by doing the NuGetAuditMode assigning in the "inner-build", which means that
+            // different inner builds can have different values. So, if any of the values is "all", then we use it.
+            // Otherwise, we can fall back to our previous behavior.
+            static string? GetAuditMode(IReadOnlyList<IVsTargetFrameworkInfo4> tfms)
+            {
+                ImmutableArray<string?> auditMode = GetNonEvaluatedPropertyOrNull(tfms, ProjectBuildProperties.NuGetAuditMode, s => s);
+
+                foreach (var value in auditMode)
+                {
+                    if (string.Equals(value, "all", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return value;
+                    }
+                }
+
+                return GetSingleNonEvaluatedPropertyOrNull(auditMode, ProjectBuildProperties.NuGetAuditMode);
```

Comment:
I wonder if we should always prefer the stricter ones if the audit modes are different.

Reply:
audit mode only has two choices, `all`, or `direct`. So, in practice it will always select the stricter, assuming that `all` is specified in at least one TFM.

But if the customer sets the value to `direct` in one TFM, and `banana` is another, then they'll get the error that it's not consistent.  It's a very unlikely scenario though, as the customer would need to conditionally set audit mode, and to a value that's neither of the valid options.

Reply:
In my head, I had combined NuGetAudit and NuGetAuditMode into one. 

So I thought it's all, direct or none/disable. 

Ignore this, you're absolutely right. 
I think the extra behavior is good.